src/figuresClasses.ts

```diff
@@ -1,19 +1,111 @@
+// import { assertAwaitExpression } from '@babel/types';
+
 export interface Figure {
+  shape: string;
+
+  color: string;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  public shape: string;
 
+  constructor(color: string, a: number, b: number, c: number, shape: string) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.shape = shape;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Negative sides!');
+    }
+
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      throw new Error('Not A triangle!');
+    }
+  }
+
+  getArea(): number {
+    const size = (this.a + this.b + this.c) / 2;
+
+    return Math.floor((Math.sqrt(size
```

Comment:
seems legit to introduce a rounding helper function to use it 3 times in your code

Reply:
Not applied , please apply this comment